### PR TITLE
Creating first tests for cmd project

### PR DESCRIPTION
**Pull request by RBlafford**
- Closing pull request for bugfix/graph_issues branch -- this branch branches from it.

### CODE REVIEW

File: concord_cli/deploy.py [hunk 1]
```diff
@@ -62,18 +62,28 @@
...
+
+    all_keys = list(valid_keys)
+    all_keys.extend(reqs)
+
     for k in dictionary:
-        if k not in valid_keys:
+        if k not in all_keys:
             parser.error("Key is not a valid concord request key: " + str(k))
 
+    for k in reqs:
```
Comment (bigs) [inline concord_cli/deploy.py]: nice :+1:

File: concord_cli/deploy.py [hunk 2]
```diff
@@ -137,7 +138,10 @@ def not_on_black_list(path):
     # iterate through a set of subdirectories, filtering by blacklist
     def iterate_files(path):
         if os.path.isfile(path):
-            return [path]
+            if not_on_black_list(path):
```
Comment (bigs) [inline concord_cli/deploy.py]: yep i still feel dumb for this one

great catch!

File: concord_cli/print_graph.py [hunk 3]
```diff
@@ -56,6 +57,16 @@ def print_single_stream(s):
 def print_streams(comp):
     return json.dumps(comp, default=lambda o: o.__dict__, indent=4)
 
+def print_edge(comp1, comp2, dot):
+    for streamMetadata in comp1.istreams:
+        if streamMetadata.name in map(attrgetter('name'), comp2.ostreams):
+            for node1 in comp1.nodes:
+                for node2 in comp2.nodes:
+                    print "EDGE!: ", node2.taskId, " -> ", node1.taskId
```
Comment (bigs) [inline concord_cli/print_graph.py]: remove print maybe?

File: concord_cli/print_graph.py [hunk 4]
```diff
@@ -56,6 +57,16 @@ def print_single_stream(s):
...
+                    dot.node(node1.taskId, print_streams(node1))
+                    dot.node(node2.taskId, print_physical(node2))
+                    dot.edge(node2.taskId, node1.taskId,
+                             label=print_single_stream(streamMetadata))
 
 def printdot(options):
```
Comment (bigs) [inline concord_cli/print_graph.py]: for consistency, `print_dot`

**Comment by bigs** [comment]
few cosmetic comments, otherwise LGTM!

File: tests/test_deploy.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        # A dictionary must have all of the required keys
+        test_config_dict = req_dict.copy()
+        test_config_dict.pop("compress_files")
+        self.assertRaises(RuntimeError,
+                          validate_json_raw_config,
+                          test_config_dict,
+                          self.stub)
+
+        # A dictionary with only the required keys passes
+        validate_json_raw_config(req_dict, self.stub)
```
Comment (senior7515) [inline tests/test_deploy.py]: maybe check if is a json file at all.

File: tests/testing_utilities.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import os
+
+def test_filepath(filepath):
+    path = os.path.dirname(os.path.realpath(__file__))
+    return path + '/' + filepath
+
+def create_temporary_file(path):
+    """
```
Comment (senior7515) [inline tests/testing_utilities.py]: :+1:  for comments.

**Comment by senior7515** [comment]
@RBlafford LGTM.

**Comment by senior7515** [comment]
Feel free to merge.

**Pull request merged**